core/node/base_token_adjuster/src/base_token_ratio_persister.rs

```diff
@@ -112,4 +137,95 @@ impl BaseTokenRatioPersister {
 
         Ok(id)
     }
+
+    async fn send_ratio_to_l1(&self, api_ratio: BaseTokenAPIRatio) -> anyhow::Result<()> {
+        let fn_set_token_multiplier = self
+            .chain_admin_contract
+            .function("setTokenMultiplier")
+            .context("`setTokenMultiplier` function must be present in the ChainAdmin contract")?;
+
+        let calldata = fn_set_token_multiplier
+            .encode_input(
+                &(
+                    Token::Address(self.diamond_proxy_contract_address),
+                    Token::Uint(api_ratio.numerator.get().into()),
+                    Token::Uint(api_ratio.denominator.get().into()),
+                )
+                    .into_tokens(),
+            )
+            .context("failed encoding `setTokenMultiplier` input")?;
+
+        let nonce = (*self.eth_client)
+            .as_ref()
+            .nonce_at_for_account(
+                self.token_multiplier_setter_account_address,
+                BlockNumber::Pending,
+            )
+            .await
+            .with_context(|| "failed getting transaction count")?
+            .as_u64();
+
+        let gas_price = (*self.eth_client)
+            .as_ref()
+            .get_gas_price()
+            .await
+            .with_context(|| "failed getting gas price")?
+            .as_u64()
+            * 2;
+
+        let base_fee_per_gas = self.gas_adjuster.as_ref().get_base_fee(0);
+        let priority_fee_per_gas = self.gas_adjuster.as_ref().get_priority_fee();
+
+        let options = Options {
+            gas: Some(U256::from(self.config.max_tx_gas)),
+            nonce: Some(U256::from(nonce)),
+            gas_price: Some(gas_price.into()),
+            max_fee_per_gas: Some(U256::from(base_fee_per_gas + priority_fee_per_gas)),
+            max_priority_fee_per_gas: Some(U256::from(priority_fee_per_gas)),
+            ..Default::default()
+        };
+
+        let signed_tx = self
+            .eth_client
+            .sign_prepared_tx_for_addr(
+                calldata,
+                self.chain_admin_contract_address.unwrap(),
+                options,
+            )
+            .await
+            .context("cannot sign a `setTokenMultiplier` transaction")?;
+
+        let hash = (*self.eth_client)
+            .as_ref()
+            .send_raw_tx(signed_tx.raw_tx)
+            .await
+            .context("failed sending `setTokenMultiplier` transaction")?;
```

Comment:
are you going to resend tx, if it stuck?

Reply:
Good point - added configurable retries